### DISCUSSION TITLE
Longitude and Latitude not returned

### DISCUSSION THREAD

**Discussion by HansKappert** [tdorssers/TeslaPy#148] [Q&A] [answered]
Am I the only one that no longer receives those fields in driving state?

**Comment by selsrog** [comment]
I'm still getting them as before...

**Reply by IMgoRt** [reply]
23.38.x removes that from the steaming API, it needs a code change to source the data from vehicle

**Reply by HansKappert** [reply]
Can you elaborate on that. I don't understand

**Reply by IMgoRt** [reply]
Musky wants to monetise our access to the API
On car software versions 2023.38 the streaming API no longer provides location data
The .py uses this data, location data can still be found from a vehicle call

The developer will need to update the code you're using to read location form vehicle

**Reply by HansKappert** [reply]
Ok, so I need to do a "vehicle call".  Are there two API's then? One I am using (streaming API as you call it) and some other? If so, can you point me to any documentation?

**Comment by priitka** [comment]
Same problem here after updating to 2023.38.6. This seems to be related to Location Sharing change where icon appears on touchscreen when an app requests location data.
I assume that location data was moved to separate stream and it is no longer available over REST API

**Reply by priitka** [reply, marked as answer]
found solution to the problem: you need to add endpoints parameter with location_data to vehicle_data request. f.ex.
/api/1/vehicles/XXXXXXXXXvehicle_data?endpoints=location_data%3Bcharge_state%3Bclimate_state%3Bvehicle_state

**Reply by HansKappert** [reply]
Thanks. Indeed changing line 34 of endpoints.json to:
`    "URI": "api/1/vehicles/{vehicle_id}/vehicle_data?endpoints=location_data%3Bcharge_state%3Bclimate_state%3Bvehicle_state%3Bvehicle_config%3Bgui_settings",`
fixed it for me.
There are however other things changed as well. Using the menu.py, I got also an error trying to see the charging history.
But thanks all for your help.

**Reply by dmaunder** [reply]
I had to make that change on line 39, ie VEHICLE_DATA, not line 34, which is VEHICLE_DATA_LEGACY, and then it worked fine for me.

**Comment by doubledrat** [comment]
this fix worked for me too

**Comment by robharman** [comment]
@tdorssers the endpoints fix in  https://github.com/tdorssers/TeslaPy/commit/38ff7da6e5a7aa1fcd5d39b84aa1d89c9cff50b5 fix this issue for me, so thanks for that! But, the build available in pypi hasn't been updated to include it.

Any chance you can update the package?
